state_nobus.c

```diff
@@ -0,0 +1,93 @@
+#include "config.h"
+#include "socketcand.h"
+#include "include/linux/canctl.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+
+void state_nobus(void)
+{
+	int i;
+	char buf[MAXLEN], op;
+
+	if(previous_state != STATE_NO_BUS) {
+		strcpy(buf, "< hi >");
+		send(client_socket, buf, strlen(buf), 0);
+		tcp_quickack(client_socket);
+		previous_state = STATE_NO_BUS;
+	}
+
+	/* client has to start with a command */
+	i = receive_command(client_socket, (char *) &buf);
```

Comment:
I think the cast is not needed

```suggestion
	i = receive_command(client_socket, &buf);
```

Reply:
Whithout the cast you receive a warning `../socketcand.h:57:39: note: expected ‘char *’ but argument is of type ‘char (*)[8290]’`.

I also inherited this from the old nobus implementation, but I think this was the reason behind the cast.

Reply:
You're right, fixed